x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/SqlDataTypes.java

```diff
@@ -458,8 +471,8 @@ public static int defaultPrecision(DataType dataType) {
         if (dataType == TEXT) {
             return 32766;
         }
-        if (dataType == DATETIME) {
-            return 3;
+        if (isDateTime(dataType)) {
+            return 9;
```

Comment:
I thought that since for the end user (driver, etc.) the 2 types are transparent and only sees one, would go for the maximum precision. Now we can differentiate the 2 types all the way if we want (need to change the type convertion functionality so that DATETIME + DATETIME_NANOS return DATETIME_NANOS as common type). But going forward we want to definitely merge the 2 types and so I chose the maximum potential precision. @bpintea Would it be a problem if a nano precision timestamp is returned but the precision we show is only `3` for the odbc driver?

Reply:
> Would it be a problem if a nano precision timestamp is returned but the precision we show is only 3 for the odbc driver?

I don't know for sure, but I believe not to be an issue.
The client can ask the data through the driver two ways:
 * as a C-struct, where the sub-second part is an unsigned long. I guess this wouldn't pose an issue (the decimal representation has more non-zeros then expected).
 * as a string, where the client would parse it itself. This could potentially pose issues with the assumed format of that string. But this _should_ be the same as returning a double with less decimals than the maximum precision.

Reply:
Imho, I'd leave it with 9 (the maximum precision) since we won't have a way to distinguish between millis & nanos precision when the types are merged.